test/compile/mark/bar.test.ts

```diff
@@ -345,15 +345,15 @@ describe('Mark: Bar', () => {
     it('should draw bar with y and y2', () => {
       expect(props.y2).toEqual([
         fieldInvalidTestValueRef(y, 'y'),
-        {scale: 'y', field: 'bin_maxbins_10_Horsepower', offset: {signal: 'a'}}
+        {scale: 'y', field: 'bin_maxbins_10_Horsepower', offset: {signal: '0.5 + (a + -0.5)'}}
       ]);
       expect(props.y).toEqual([
         fieldInvalidTestValueRef(y, 'y'),
         {
           scale: 'y',
           field: 'bin_maxbins_10_Horsepower_end',
           offset: {
-            signal: '1 + a'
+            signal: '0.5 + (a + 0.5)'
```

Comment:
This looks worse than before. Any chance you can simplify the expression?

Reply:
i spent a good amount of time trying, but the complexity wasn't worth it imo. Here's the non-signal expression:

`translate + (reverse ? -offset - spacingOffset : +offset + spacingOffset)`

The problem I ran into is that there are 3 values (`translate`, `offset`, and `spacing`) and 2 of them can be signals (`translate` and `offset`). You'd have to conditionally combine `spacing` and the non-signal (if there is one). On top of that, `offset` and `spacing` are modified (turned negative) if `reverse` is true, but `translate` is not.

Instead, this PR takes the approach of the formula

`translate +  (reverse ? -1 : 1) * (offset + spacing)` and collapses when values are missing. 

But let me know if you see a better solution I might have overlooked.

Reply:
I guess what's nice is that we use the same expressions.